### PR TITLE
SystemTests: Secretutils encode/decode passwords method

### PR DESCRIPTION
**Pull request by michalxo**
Signed-off-by: Michal T <mtoth@redhat.com>

- Refactoring ST

Use encoding method for plain passwords (added decode method as well)

- [x] Make sure all tests pass

### CODE REVIEW

**Comment by scholzj** [comment]
Just FYI: You know you can also use `.withStringData()` in the SecretBuilder when creating the secret to build the secret from plaintext fields? (to be clear, I do not have an issue with the current code you have here => just wondered if you know abotu it)

**Comment by michalxo** [comment]
No, that's new for me, thank you, great to know. 
Also, there is 0 usages of `withStringData` in whole systemtests (or whole project).

**Comment by michalxo** [comment]
@strimzi-ci run tests test_only profile=regression testcase=OauthPlainIsolatedST,OauthAuthorizationIsolatedST,OauthScopeIsolatedST,OauthTlsIsolatedST

File: systemtest/src/main/java/io/strimzi/systemtest/utils/kubeUtils/objects/SecretUtils.java [hunk 1]
```diff
@@ -181,4 +181,12 @@ public static String annotateSecret(String namespaceName, String resourceName, S
                 .out()
                 .trim();
     }
+
+    public static String encodeStringPassword(String plainPassword) {
+        return new String(Base64.getEncoder().encode(plainPassword.getBytes()), StandardCharsets.US_ASCII);
```
[the remark this answers is not in the document]
Comment (michalxo) [reply systemtest/src/main/java/io/strimzi/systemtest/utils/kubeUtils/objects/SecretUtils.java:186]: Not possible sadly. 
The bigger problem I see are some weirdly failing tests after this change, which I don't understand at the moment.

**Comment by michalxo** [comment]
@strimzi-ci run tests test_only profile=regression testcase=OauthPlainIsolatedST,OauthAuthorizationIsolatedST,OauthScopeIsolatedST,OauthTlsIsolatedST,SecurityST

**Comment by michalxo** [comment]
Seems like (at least) following 2 tests are failing (also on main)
```
SecurityST.testAclWithSuperUser
SecurityST.testAclRuleReadAndWrite
```
I will have a look at them and fix them in separate PR.

**Comment by michalxo** [comment]
Wau, all passed. Probably good to merge then? :-)

**Comment by michalxo** [comment]
There was an error in pipeline post-job - maven-cache phase. But all tests have passed fine.
``` 
@azure-pipelines azure-pipelines / regression (regression-bundle VII. - remaining system tests regression_all_remaining)
Build log #L43
The server is busy.  Please try again later.
```
```[WARNING] Tests run: 62, Failures: 0, Errors: 0, Skipped: 2, Flakes: 1```

I would say AZP passed. Feel free to double-check please.

**Review by Frawless — APPROVED** [review]

**Comment by Frawless** [comment]
I confirm that tests are passing on AZP and the problem was with some artifact publishing.

**Review by im-konge — APPROVED** [review]

**Pull request merged**